RN 0.76 Vanilla example can't run

### Description

RN 0.76 just released. At first, I tested with [my library](https://github.com/pnthach95/react-native-root-view-background) but it can't run. Then, I created new library and vanilla example can't run.

Android Vanilla example log:

```bash
> Task :app:invokeLibraryCodegen FAILED
ℹ Building target codegen
ℹ Cleaning up previous iOS codegen native code at ios/generated
ℹ Cleaning up previous Android codegen native code at android/generated
✖ 
17 actionable tasks: 17 executed

info 💡 Tip: Make sure that you have set up your development environment correctly, by running npx react-native doctor. To read more about doctor command visit: https://github.com/react-native-community/cli/blob/main/packages/cli-doctor/README.md#doctor 

/Volumes/code/Projects/awesome-library/node_modules/react-native-builder-bob/lib/targets/codegen.js:48
    throw new Error('Failed generate the codegen files.');
          ^

Error: Failed generate the codegen files.
    at build (/Volumes/code/Projects/awesome-library/node_modules/react-native-builder-bob/lib/targets/codegen.js:48:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async buildTarget (/Volumes/code/Projects/awesome-library/node_modules/react-native-builder-bob/lib/index.js:421:7)

Node.js v20.12.1

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:invokeLibraryCodegen'.
> Process 'command 'npx'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 8s
error Failed to install the app. Command failed with exit code 1: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
/Volumes/code/Projects/awesome-library/node_modules/react-native-builder-bob/lib/targets/codegen.js:48 throw new Error('Failed generate the codegen files.'); ^ Error: Failed generate the codegen files. at build (/Volumes/code/Projects/awesome-library/node_modules/react-native-builder-bob/lib/targets/codegen.js:48:11) at process.processTicksAndRejections (node:internal/process/task_queues:95:5) at async buildTarget (/Volumes/code/Projects/awesome-library/node_modules/react-native-builder-bob/lib/index.js:421:7) Node.js v20.12.1 FAILURE: Build failed with an exception. * What went wrong:
Execution failed for task ':app:invokeLibraryCodegen'.
> Process 'command 'npx'' finished with non-zero exit value 1 * Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org. BUILD FAILED in 8s.
info Run CLI with --verbose flag for more details.
```

iOS Vanilla example log:

```bash
ℹ Building target codegen
ℹ Cleaning up previous iOS codegen native code at ios/generated
ℹ Cleaning up previous Android codegen native code at android/generated
✖ 
/Volumes/code/Projects/awesome-library/node_modules/react-native-builder-bob/lib/targets/codegen.js:48
    throw new Error('Failed generate the codegen files.');
          ^

Error: Failed generate the codegen files.
    at build (/Volumes/code/Projects/awesome-library/node_modules/react-native-builder-bob/lib/targets/codegen.js:48:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async buildTarget (/Volumes/code/Projects/awesome-library/node_modules/react-native-builder-bob/lib/index.js:421:7)
```

EDIT: Workaround is open `node_modules/react-native-builder-bob/lib/targets/codegen.js` and make change

```patch
-    await (0, _spawn.spawn)('npx', ['react-native', 'codegen'], {
+    await (0, _spawn.spawn)('npx', ['@react-native-community/cli', 'codegen'], {
```

### Packages

- [X] create-react-native-library
- [ ] react-native-builder-bob

### Selected options

✔ What is the name of the npm package? … react-native-awesome-library
✔ What is the description for the package? … test rn76
✔ What is the name of package author? … Pham Ngoc Thach
✔ What is the email address for the package author? … thachxyz123@gmail.com
✔ What is the URL for the package author? … https://github.com/pnthach95
✔ What is the URL for the repository? … https://github.com/pnthach95/react-native-awesome-library
✔ What type of library do you want to develop? › Turbo module with backward compat
✔ Which languages do you want to use? › Kotlin & Objective-C
✔ What type of example app do you want to create? › Vanilla

### Link to repro

https://github.com/pnthach95/react-native-awesome-library

### Environment

  OS: macOS 15.0.1
  CPU: (8) arm64 Apple M1
  Node:
    version: 20.12.1
    path: ~/.nvm/versions/node/v20.12.1/bin/node
  Yarn:
    version: 3.6.4
    path: ~/.nvm/versions/node/v20.12.1/bin/yarn
  npm:
    version: 10.8.2
    path: ~/.nvm/versions/node/v20.12.1/bin/npm

--- a/packages/create-react-native-library/src/index.ts
+++ b/packages/create-react-native-library/src/index.ts
@@ -790,8 +790,23 @@ async function create(_argv: yargs.Arguments<any>) {
         examplePackageJson.dependencies['react-native'];
     }
 
-    if (arch !== 'legacy' && example === 'vanilla') {
-      addCodegenBuildScript(folder, options.project.name);
+    if (example === 'vanilla') {
+      // React Native doesn't provide the community CLI as a dependency.
+      // We have to get read the version from the example app and put to the root package json
+      const exampleCommunityCLIVersion =
+        examplePackageJson.devDependencies['@react-native-community/cli'];
+      console.assert(
+        exampleCommunityCLIVersion !== undefined,
+        "The generated example app doesn't have community CLI installed"
+      );
+
+      rootPackageJson.devDependencies = rootPackageJson.devDependencies || {};
+      rootPackageJson.devDependencies['@react-native-community/cli'] =
+        exampleCommunityCLIVersion;
+
+      if (arch !== 'legacy') {
+        addCodegenBuildScript(folder, options.project.name);
+      }
     }
   }
 

--- a/packages/create-react-native-library/src/index.ts
+++ b/packages/create-react-native-library/src/index.ts
@@ -5,6 +5,7 @@ import dedent from 'dedent';
 import kleur from 'kleur';
 import yargs from 'yargs';
 import ora from 'ora';
+import assert from 'node:assert';
 import validateNpmPackage from 'validate-npm-package-name';
 import githubUsername from 'github-username';
 import prompts, { type PromptObject } from './utils/prompts';
@@ -795,7 +796,7 @@ async function create(_argv: yargs.Arguments<any>) {
       // We have to get read the version from the example app and put to the root package json
       const exampleCommunityCLIVersion =
         examplePackageJson.devDependencies['@react-native-community/cli'];
-      console.assert(
+      assert(
         exampleCommunityCLIVersion !== undefined,
         "The generated example app doesn't have community CLI installed"
       );

--- a/packages/create-react-native-library/templates/common/$package.json
+++ b/packages/create-react-native-library/templates/common/$package.json
@@ -72,6 +72,9 @@
   "devDependencies": {
     "@commitlint/config-conventional": "^17.0.2",
     "@evilmartians/lefthook": "^1.5.0",
+<% if (example === 'vanilla') { -%>
+    "@react-native-community/cli": "15.0.0-alpha.2",
+<% } -%>
     "@react-native/eslint-config": "^0.73.1",
     "@release-it/conventional-changelog": "^5.0.0",
     "@types/jest": "^29.5.5",

--- a/packages/react-native-builder-bob/src/targets/codegen.ts
+++ b/packages/react-native-builder-bob/src/targets/codegen.ts
@@ -33,7 +33,7 @@ export default async function build({ root, report }: Options) {
   }
 
   try {
-    await spawn('npx', ['react-native', 'codegen'], {
+    await spawn('npx', ['@react-native-community/cli', 'codegen'], {
       stdio: 'ignore',
     });
 

